client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -93,17 +93,39 @@ public synchronized int register(String subject, Schema schema)
   }
 
   @Override
-  public synchronized Schema getByID(int id) throws IOException, RestClientException {
-    if (idCache.containsKey(id)) {
-      return idCache.get(id);
+  public synchronized Schema getBySubjectAndID(String subject, int id)
+      throws IOException, RestClientException {
+    Map<Integer, Schema> idSchemaMap;
+    if (idCache.containsKey(subject)) {
+      idSchemaMap = idCache.get(subject);
     } else {
+      idSchemaMap = new IdentityHashMap<Integer, Schema>();
```

Comment:
Why an `IdentityHashMap`? I think you need to be careful with this due to Java's integer pool. This will work for schema IDs between -128 and 127, but after that the boxed integers won't be `==`.